src/vsr/client.zig

```diff
@@ -21,45 +21,19 @@ pub fn Client(comptime StateMachine_: type, comptime MessageBus: type) type {
         const Self = @This();
 
         pub const StateMachine = StateMachine_;
-
+        pub const DemuxerType = StateMachine.DemuxerType;
         pub const Request = struct {
             pub const Callback = *const fn (
                 user_data: u128,
                 operation: StateMachine.Operation,
-                results: []const u8,
+                results: []u8,
```

Comment:
In `vsr.Client.on_reply`, can we assert `reply.references == 1` just before 

```
                // The message is the result of raw_request(), so invoke the user callback.
                inflight.callback(
                    inflight.user_data,
                    inflight_vsr_operation.cast(StateMachine),
                    reply.body(),
                );
```

? I'm not sure if it will work... there might be an extra reference in the message bus, but it is worth a try.

And either way, I would add a comment to that callback invocation warning that the callback is allowed to mutate the reply body in-place.

Reply:
To guarantee no other references, should vsr copy the reply body to its own internal buffer and pass that to the callback?

Reply:
No, I was just wondering if we already had a guarantee of no other references. I don't think it is worth the overhead of copying it.